### PR TITLE
add k8s cloud provider config in windows node

### PR DESCRIPTION
**Pull request by andyzhangx**
**What this PR does / why we need it**:
I found k8s cloud provider(azure) is not configured in windows node, this is the first step to enabling k8s azure disk on windows

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/1308)

### CODE REVIEW

**Comment by andyzhangx** [comment]
@jdumars for awareness

**Comment by seanknox** [comment]
Awesome--thanks @andyzhangx! FYI @jackfrancis

**Comment by jackfrancis** [comment]
@JiangtianLi could you review this change?

File: parts/kuberneteswindowssetup.ps1 [hunk 1]
```diff
@@ -177,7 +177,7 @@ Write-KubernetesStartFiles($podCIDR)
 {
     $KubeletArgList = @("--hostname-override=`$global:AzureHostname","--pod-infra-container-image=kubletwin/pause","--resolv-conf=""""""""","--api-servers=https://`${global:MasterIP}:443","--kubeconfig=c:\k\config")
     $KubeletCommandLine = @"
-c:\k\kubelet.exe --hostname-override=`$global:AzureHostname --pod-infra-container-image=kubletwin/pause --resolv-conf="" --allow-privileged=true --enable-debugging-handlers --api-servers=https://`${global:MasterIP}:443 --cluster-dns=`$global:KubeDnsServiceIp --cluster-domain=cluster.local  --kubeconfig=c:\k\config --hairpin-mode=promiscuous-bridge --v=2 --azure-container-registry-config=c:\k\azure.json --runtime-request-timeout=10m
+c:\k\kubelet.exe --hostname-override=`$global:AzureHostname --pod-infra-container-image=kubletwin/pause --resolv-conf="" --allow-privileged=true --enable-debugging-handlers --api-servers=https://`${global:MasterIP}:443 --cluster-dns=`$global:KubeDnsServiceIp --cluster-domain=cluster.local  --kubeconfig=c:\k\config --hairpin-mode=promiscuous-bridge --v=2 --azure-container-registry-config=c:\k\azure.json --runtime-request-timeout=10m  --cloud-provider=azure --cloud-config=c:\k\azure.json
```
Comment (JiangtianLi) [inline parts/kuberneteswindowssetup.ps1]: Does this switch work for all versions from 1.5 - 1.7?

Comment (JiangtianLi) [inline parts/kuberneteswindowssetup.ps1]: If not, please add into the if conditional branch properly below. Otherwise, LGTM.

Comment (andyzhangx) [reply parts/kuberneteswindowssetup.ps1]: @JiangtianLi it works for all versions, before v1.8, cloud-provider & cloud-config is not used in k8s kubelet in windows, I also did the test, it works well.

Comment (JiangtianLi) [reply parts/kuberneteswindowssetup.ps1]: @andyzhangx So mean --cloud-provider and --cloud-config exist all for version 1.5 and up but are no-op before 1.8? I'll sign-off then. Thanks.

Comment (seanknox) [reply parts/kuberneteswindowssetup.ps1]: > So mean --cloud-provider and --cloud-config exist all for version 1.5 and up but are no-op before 1.8? 

Yep!

Comment (andyzhangx) [reply parts/kuberneteswindowssetup.ps1]: @JiangtianLi yes, and it's not before **v1.8.0**, I have another PR(azure disk on windows, https://github.com/kubernetes/kubernetes/pull/51252) which depends on this config,  that's why I created this acs-engine config change PR. And https://github.com/kubernetes/kubernetes/pull/51252 would be likely into v1.8.1, so let's say k8s v1.8.1 depends on this cloud config.

**Comment by andyzhangx** [comment]
@JiangtianLi @dmitsh Is there something wrong with the CircleCI test?

**Comment by andyzhangx** [comment]
@JiangtianLi @dmitsh Is there something wrong with the CircleCI test? This PR is for Windows k8s GA.

**Comment by JiangtianLi** [comment]
@andyzhangx Can you rebase and update your branch? There was some issue with windows zip before.

**Comment by JiangtianLi** [comment]
@andyzhangx Can you merge with master and push again? Hopefully it'll get the tests pass. AFAIK, the tests are in a good state recently. If you still see failure, please let @jackfrancis and me know.

**Comment by andyzhangx** [comment]
@JiangtianLi I have rebased again.

**Comment by JiangtianLi** [comment]
@andyzhangx approved. Let's see how the tests go. Thanks.

**Comment by JiangtianLi** [comment]
@andyzhangx The test failed due to it failed to curl [simple web service](https://github.com/JiangtianLi/acs-engine/blob/master/docs/kubernetes.windows.md#create-your-first-kubernetes-service). I manually deployed two 1.6 hybrid cluster (with both windows node and linux node). The acs-engine HEAD doesn't have the failure while the one with your PR has the failure. While I am looking into it, could you please also take a look on your side?

**Comment by JiangtianLi** [comment]
@andyzhangx  I found the issue. Please add `cloud-provider=azure --cloud-config=c:\k\azure.json` also to `$KubeletArgList`

The test failure is that the master can't reach windows POD IP, cluster IP, service IP.

The root cause is that on Windows node, it initializes by starting kubelet to get podCIDR. Then later it starts kubelet as service. The first time it starts kubelet, it has no additional parameter you added. Therefore it causes the podCIDR for the windows node to change later and therefore in master's route table, it doesn't match the network setup on windows node.

I manually created several clusters and verified the fix. Please fix and re-submit.

**Comment by seanknox** [comment]
giphy 10

Kudos @andyzhangx and @JiangtianLi!

**Pull request merged**